server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadataBatch.java

```diff
@@ -0,0 +1,535 @@
+
+        public NodesStoreFilesMetadataBatch(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        public NodesStoreFilesMetadataBatch(
+            ClusterName clusterName,
+            List<NodeStoreFilesMetadataBatch> nodes,
+            List<FailedNodeException> failures
+        ) {
+            super(clusterName, nodes, failures);
+        }
+
+        @Override
+        protected List<NodeStoreFilesMetadataBatch> readNodesFrom(StreamInput in) throws IOException {
+            return in.readList(NodeStoreFilesMetadataBatch::new);
+        }
+
+        @Override
+        protected void writeNodesTo(StreamOutput out, List<NodeStoreFilesMetadataBatch> nodes) throws IOException {
+            out.writeList(nodes);
+        }
+    }
+
+    /**
+     * The metadata for the node store files
+     *
+     * @opensearch.internal
+     */
+    public static class NodeStoreFilesMetadata {
+
+        private StoreFilesMetadata storeFilesMetadata;
+        private Exception storeFileFetchException;
+
+        public NodeStoreFilesMetadata(StoreFilesMetadata storeFilesMetadata) {
+            this.storeFilesMetadata = storeFilesMetadata;
+            this.storeFileFetchException = null;
+        }
+
+        public NodeStoreFilesMetadata(StreamInput in) throws IOException {
+            storeFilesMetadata = new StoreFilesMetadata(in);
+            if (in.readBoolean()) {
+                this.storeFileFetchException = in.readException();
+            } else {
+                this.storeFileFetchException = null;
+            }
+        }
+
+        public NodeStoreFilesMetadata(StoreFilesMetadata storeFilesMetadata, Exception storeFileFetchException) {
+            this.storeFilesMetadata = storeFilesMetadata;
+            this.storeFileFetchException = storeFileFetchException;
+        }
+
+        public StoreFilesMetadata storeFilesMetadata() {
+            return storeFilesMetadata;
+        }
+
+        public static NodeStoreFilesMetadata readListShardStoreNodeOperationResponse(StreamInput in) throws IOException {
+            return new NodeStoreFilesMetadata(in);
+        }
+
+        public void writeTo(StreamOutput out) throws IOException {
+            storeFilesMetadata.writeTo(out);
+            if (storeFileFetchException != null) {
+                out.writeBoolean(true);
+                out.writeException(storeFileFetchException);
+            } else {
+                out.writeBoolean(false);
+            }
+        }
+
+        public Exception getStoreFileFetchException() {
+            return storeFileFetchException;
+        }
+
+        @Override
+        public String toString() {
+            return "[[" + storeFilesMetadata + "]]";
+        }
+    }
+
+    /**
+     * NodeRequest class is for deserializing the  request received by this node from other node for this transport action.
+     * This is used in {@link TransportNodesAction}
```

Comment:
nit: Documentation misleading here, how it used in TransportNodesAction. Or is it I am missing something here?

Reply:
As we are extending TransportNodesAction and the same class uses generics our specified classes for request and response will be used while performing  the transport call.
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/nodes/TransportNodesAction.java